File: benchmarks/fixtures/kernels.fs
module Benchmark

let Add a b = a + b

let rec Fibonacci n =
    if n < 2 then n
    else Fibonacci (n - 1) + Fibonacci (n - 2)

let Checksum n seed =
    if n <= 0 then seed
    else
        let mutable value = seed
        for i = 0 to n - 1 do
            value <- (value ^^^ (value <<< 5)) + (i ^^^ (value >>> 2))
        value

File: benchmarks/fixtures/kernels.go
package main

//go:wasmexport Add
func Add(a, b int32) int32 { return a + b }

//go:wasmexport Fibonacci
func Fibonacci(n int32) int32 {
	if n < 2 {
		return n
	}
	return Fibonacci(n-1) + Fibonacci(n-2)
}

//go:wasmexport Checksum
func Checksum(n, seed int32) int32 {
	value := seed
	for i := int32(0); i < n; i++ {
		value = (value ^ (value << 5)) + (i ^ (value >> 2))
	}
	return value
}

func main() {}
